src/pal/src/thread/thread.cpp

```diff
@@ -573,7 +568,22 @@ CorUnix::InternalCreateThread(
         palError = ERROR_INVALID_PARAMETER;
         goto EXIT;
     }
-    
+
+    if (dwStackSize != 0)
+    {
+        // Some systems require the stack size to be aligned to the page size
+        DWORD alignedStackSize = static_cast<DWORD>(ALIGN_UP(dwStackSize, OS_PAGE_SIZE));
+        if (alignedStackSize < dwStackSize)
+        {
+            // When coming here from the public API surface, the incoming value is originally a nonnegative signed int32, so
+            // this shouldn't happen
+            ASSERT("Couldn't align the requested stack size (%u) to the page size\n", dwStackSize);
```

Comment:
On 32-bit platforms, DWORD and size_t have the same size and both are unsigned, so it could still overflow even after casting to size_t. I could make the check specific for 32-bit.